### DISCUSSION TITLE
GTFS Trip-Modifications support

### DISCUSSION THREAD

**Discussion by scrudden** [TheTransitClock/transitime#276] [Ideas]
Is it worth adding support for GTFS Trip-Modifications to TheTransitClock?

https://github.com/google/transit/pull/403

**Comment by jamelkhiareddine** [comment]
Yes.

**Comment by larbizard** [comment]
Hi @scrudden It's worth adding support to GTFS Trip-Modifications to the TransitClock.
I would like to contribute. Do you have any list of tasks to begin with to add this new GTFS feature ?

**Comment by scrudden** [comment]
Detours have been talked about before.

https://github.com/TheTransitClock/transitime/issues/152

**Comment by MANISH-K-07** [comment]
Hey @scrudden , 
I am **Manish K**, an undergrad sophomore pursuing Computer Science Engineering in India.
A brief intro about me would be that I basically breathe Java and have pretty good open source experience. I love challenges and am a very quick learner. This isn't ideally related but my dad works for backend development of Penske, so I also have a good understanding of
General transit feed and why we use it.

Coming to the point, I have been going through the docs and discussions here and I see that #152 has been open since the past 5 years despite the ROI being monumentally high !! Wouldn't it be amazing if this problem could be solved this summer ?? I would love to work on this as part of gsoc and get a fully functional detection model up and running by end of gsoc period.

Honestly, I regret not having stumbled upon this project earlier. I was browsing for a good project to work on for this summer when I found OTSF. Since its already too late to showcase my skills through contributions here, I have no other option than sharing my experience of working with another widely used static code analysis tool called [Checkstyle](https://checkstyle.org/). I am an active contributor at this Java based project and have about 70 successfully merged PRs.

- https://github.com/checkstyle/checkstyle/commits?author=MANISH-K-07
- https://github.com/checkstyle/contribution/commits?author=MANISH-K-07

I have worked on many bug fixes, documentation enhancements, migrations, bulk updates and a lot more. I really hope this would suffice to showcase my experience here and convince you regarding my claims about the skillset I possess.

Cheers,
Manish K (https://github.com/MANISH-K-07)

**Comment by MANISH-K-07** [comment]
Also, I'm figuring out the best possible POA to approach and tackle this project. As I'm not yet very familiar with the codebase of this project, any and all insights would be highly appreciated. Just to be sure, like I mentioned, I'm a very quick learner and will grasp in a jiffy 😄 

PS : I have sent a join request at the mailing list and am looking forward to diving in.

**Comment by MANISH-K-07** [comment]
@scrudden , ping please

**Comment by scrudden** [comment]
Sent email.

**Reply by MANISH-K-07** [reply]
> Sent email.

Thanks a lot for responding.
Could you please excuse me and look for my response to your email. I'll be waiting for your feedback there
